### PR TITLE
Wacky Worlds/Time Twister Vehicle Enhancement Project

### PR DESCRIPTION
**Pull request by Fred-1044276**
I have made numerous changes and fixes to the Wacky Worlds and Time Twister vehicles, including but not limited to:

☆ Added Japanese (ja-JP) translations and names for the vehicles (with at least the name and capacity included). Here are several examples of the Japanese names I added:

 - Dragonfly Cars (Mini Suspended Coaster) → 蜻蛉車両
 - Surfing Trains (Steel Twister) → サーフボードトレイン (lit. "Surfboard Trains")
 - Diamond Ride (Twist) → スクランブルダイヤモンド (lit. "Scrambled Diamonds")
 - Fighting Kite Ride (Twist) → 戦闘凧ツイスター (lit. "Fighting Kite Twister")
 - Neptune Ride (Twist) → ネプチューンのツイスター (lit. "Neptune's Twister")
 - Bullet Trains (Corkscrew Roller Coaster) → 新幹線 (lit. "Shinkansen")
 - 1950's Rockets (Suspended Swinging Coaster) → 1950年代のロケット車両 (lit. "1950s Rocket Cars")

☆ Fixed/simplified the descriptions for the 1950's Rockets (Suspended Swinging Coaster) and BarnStorming Cars (Inverted Impulse Coaster), along with many others.

☆ The following vehicles now have new names (along with several others):
 - Hot Rod Cars → Hot Rod Trains
 - Russian Cars → Trabant Cars (for consistency with other languages; they also look like the Trabant 601)
 - BarnStorming Cars → Barnstorming Trains (less weird capitalization and more consistency with the Impulse Trains)

☆ Added hasShelter flags for the Black Cabs and Hovercars (Car Ride), Police Car Trains (Looping Roller Coaster), Russian Cars (Mini Roller Coaster), and Manta Ray Boats and Flying Boats (Water Coaster)

☆ Increased maxCarsPerTrain (this includes any empty cars) for the following vehicles:
 - Mine Cart Trains (Wooden Roller Coaster): 6 → **7** (for RCT1 parity)
 - Police Car Trains (Looping Roller Coaster): 7 → **8** (for RCT1 parity)
 - TGV Trains (LIM Launched Roller Coaster): 7 → **8** (for LIM Trains parity)
 - Surfing Trains (Steel Twister): 7 → **9** (for RCT1 parity; already covered in #64)

☆ Increased minCarsPerTrain for Anaconda Trains (Hypercoaster) from 2 to 3

☆ Fixed capacity typos for the expansion Mini Suspended Coaster vehicles (Sputnik Cars, Dragonfly Cars, Hoverboards)

☆ Manta Ray Boats (Water Coaster) now slow down and make splashing sounds on water sections again (already covered in #65)

☆ "numEmptyCars" flag is now present for the Anaconda Trains (Hypercoaster) and TGV Trains (LIM Launched Roller Coaster); both have this flag set to 1, making the rear tail vehicle no longer count as part of the number of cars per train (which now caps at 7 for both train types) — WARNING: This could break existing saved games, scenarios, and track designs that use any one of these trains (e.g. European Extravaganza from Wacky Worlds)

This pull request should be combined with #55, #58 (which should be addressed to fix the missing diagonal sprites for Mini Suspended Coaster vehicles), and #66.

### CODE REVIEW

File: objects/rct2tt/ride/rct2.tt.1920sand.json [hunk 1]
```diff
@@ -32,22 +32,24 @@
...
             "cs-CZ": "Art Deco stánek",
             "pl-PL": "Kiosk art deco",
             "ru-RU": "Палатка Арт-Деко"
         },
         "description": {
-            "en-GB": "A stall selling Noodles and Fresh Lemonade",
+            "en-GB": "A stall selling noodles and fresh lemonade",
             "fr-FR": "Un restaurant proposant des pâtes et de la limonade",
-            "de-DE": "Ein Stand, der Nudeln und frische Limonade verkauft",
+            "de-DE": "Ein stand, der nudeln und frische limonade verkauft",
```
Comment (AaronVanGeffen) [inline objects/rct2tt/ride/rct2.tt.1920sand.json]: This should be reverted; German uses uppercase to denote nouns.

File: objects/rct2tt/ride/rct2.tt.ganstrcr.json [hunk 2]
```diff
@@ -100,6 +100,7 @@
             "sv-SE": "Gangsterbil",
             "ko-KR": "갱스터 차량",
             "zh-TW": "盜匪車雲霄飛車",
+            "ja-JP": "ギャングスターカー"
```
Comment (AaronVanGeffen) [inline objects/rct2tt/ride/rct2.tt.ganstrcr.json]: Missing trailing comma:
```suggestion
            "ja-JP": "ギャングスターカー",
```

File: objects/rct2ww/ride/rct2.ww.minecart.json [hunk 3]
```diff
@@ -78,6 +79,7 @@
             "sv-SE": "Berg- och dalbanevagnar av trä med stoppade säten och säkerhetsbälten",
             "zh-CN": "有装填物的座位及护膝吧装置的木制云霄飞车",
             "cs-CZ": "Dřevěná horská dráha s vozy s polstrovanými sedadly a opěrkou v klíně",
+            "ja-JP": "膝の安全バーが付いた、パッド入りの座席を持つ木製のコースターですが、鉱山カートのような外観があります"
```
Comment (AaronVanGeffen) [inline objects/rct2ww/ride/rct2.ww.minecart.json]: Missing trailing comma:
```suggestion
            "ja-JP": "膝の安全バーが付いた、パッド入りの座席を持つ木製のコースターですが、鉱山カートのような外観があります",
```

File: objects/rct2tt/ride/rct2.tt.1920sand.json (hunk 1 above)
Comment (Gymnasiast) [reply objects/rct2tt/ride/rct2.tt.1920sand.json]: In general, please do not touch languages you don't speak very well.

**Comment by Gymnasiast** [comment]
Let's start off by saying that I'm happy with the effort you put in, because I'm afraid I'm going to give you a lot of homework to do. ;-)

It seems like you have merged in #55, or at least some of its changes. That's going to cause headaches with all the merge conflicts. Since #55 was earlier and is also more limited in scope, that one will be merged first. After that, you should rebase/recreate your changes on top of that.

Like Aaron already noted, this PR is very big and mixes stuff. We can't really get it reviewed like this, so we'd prefer if you split off the functional changes like the cars per train from this one. As a rule of thumb, you should create a separate PR for every point you listed.

I know this means a lot of work, but we really have to check all changes thoroughly, and having them in nice, separate PRs is going to make that a lot easier. If you do that, I think that most of your changes will get in.

**Comment by AaronVanGeffen** [comment]
It looks like you used GitHub's web-interface to create each of your commits. I imagine the prospect of having to split this PR is probably a little daunting. If you need some help with this, please do ask. You can reach us on [our Gitter channel](https://gitter.im/OpenRCT2/OpenRCT2).

**Comment by Fred-1044276** [comment]
I'm still relatively new to GitHub as I only joined this month. I might actually try to split this into multiple parts, each one based on the theme they come from (Antarctic, Prehistoric, etc.).

**Comment by Gymnasiast** [comment]
@Fred-1044276 They should be grouped on the kind of change, not the type of object. Basically, all Japanese translations should be in one PR, English in another, vehicle changes in yet another, etc.

I can split off the Japanese translations for you, but before that, I'd like you to answer Aaron's question:

> @AaronVanGeffen : Could you elaborate a little on how you settled on the Japanese translations?

Basically, we want to ensure that any Japanese translation that goes in is clear enough and grammatically correct, hence this question.

**Comment by Fred-1044276** [comment]
I derived many of my Japanese descriptions from the ones that already exist in the game (i.e. Mini Roller Coaster), as well as manually translating ride/attraction names into Japanese myself. The problem is, I can't put the JP translation fixes into a separate PR as every file I edited now has JP translations. So instead, I will separate all the files by the themeing they fall into (e.g. Asia, Prehistoric, Roaring Twenties, South America), but the vehicle fixes will come first.

**Comment by AaronVanGeffen** [comment]
Can we help? I would like to prevent you from having to use the GitHub interface for every file manually.

My suggestion would be for you to create a new PR just for the vehicle fixes. No need to split them by theme. We can (semi-)automatically rebase this PR after the vehicle fixes have been merged, so only the translations remain for review.

**Comment by Fred-1044276** [comment]
Yes, you can help. I will accept any and all help.

**Comment by Gymnasiast** [comment]
I have split off the Japanese translations: #67 
I have also split off (most of) the capacity fixes: https://github.com/OpenRCT2/objects/commit/8a4b6cae5d12ae8f9d3600fbbddf9645b9e067f1

You can leave this PR open, but please do not add anything more to it.

**Comment by AaronVanGeffen** [comment]
The Japanese WW and TT translations have been corrected and merged.

What is left to be merged from this PR?

**Comment by Fred-1044276** [comment]
Thank you for improving on my JP translations. They are much better now than when I first put them together.

One of my edits to the JSON files for the Manta Ray Boats and Flying Boats for the Water Coaster (from WW and TT respectively) added shelter to these vehicles as they do appear enclosed, but these fixes still haven't been implemented yet.

**Comment by Gymnasiast** [comment]
> One of my edits to the JSON files for the Manta Ray Boats and Flying Boats for the Water Coaster (from WW and TT respectively) added shelter to these vehicles as they do appear enclosed, but these fixes still haven't been implemented yet.

I did explain why in #65: https://github.com/OpenRCT2/objects/issues/65#issuecomment-605610657 :

> I won't add the hasShelter flags, though, because while they're technically covered (in that they have a roof), water will still come in from the front if you ride it.

**Comment by AaronVanGeffen** [comment]
I trust this can be closed, then?

**Comment by Gymnasiast** [comment]
@AaronVanGeffen There are some changes to descriptions I still want to consider.

**Comment by Wuzzy2** [comment]
I don't think it is a good idea to mix string changes with functional changes.
This PR throws in too many different things at once, and makes reviews difficult.

This is just my personal observation, I have no say over this project, and you can ignore me. But from experience I know it's much less painful when PRs are posted on a per-feature basis. It's especially painful when LOTS of unrelated things (translations, source string changes, functional changes) are thrown into one bucket, and it gets worse for huge PRs.

**Comment by AaronVanGeffen** [comment]
@Wuzzy2 Thank you. Yes, this is the conclusion @Gymnasiast and myself reached as well, and the reason effort was put into making spin-off PRs.

**Comment by Gymnasiast** [comment]
@Wuzzy2 If you had looked at this PR's comments, you'd have seen that 1) we already said that, and 2) we split up the PR because of it.

**Comment by Gymnasiast** [comment]
@Fred-1044276 Could you open a pull request for any descriptions you think still need to be changed?

**Pull request closed without merging**